Source/WebKit/UIProcess/API/haiku/WebView.cpp

```diff
@@ -62,6 +63,38 @@ const API::PageConfiguration& pageConfig)
 		fPage->drawingArea()->setSize(IntSize(rect.right - rect.left,
 		rect.top - rect.bottom));
 	}
+	BRect p(0,0,10,20);
+	paint(WebCore::IntRect(p));
 }
 
+static void drawPageBackground(const WebPageProxy* page,const BRect& rect)
+{
+	if(!page->drawsBackground())
+	return;
+	
+		
+}
+	
+
+void BWebView::paint(const IntRect& dirtyRect)
+{
+	if(dirtyRect.isEmpty())
+	{
+		fprintf(stderr,"its empty\n");
+		return;
+	}
+	fPage->endPrinting();
+	if(DrawingAreaProxyImpl* drawingArea = static_cast <DrawingAreaProxyImpl*>(fPage->drawingArea())) 
+	{
+		fprintf(stderr,"its painting\n");
+		WebCore::Region unpainted;
+		BView* surface = new BView("drawing_surface",B_WILL_DRAW);
```

Comment:
BWebView is a BView, and it's the one you should draw on. Here you are creating a new BView every time this is called, which is a huge memory leak, and you are not going to draw anything onto the BWebView itself.

Reply:
Oops im still learning so please forgive me. I do understand that it is wrong now. Uh how can i miss it out I am supposed to use the existing view to draw but im just overlaying it over and over.